lib/abr/simple_abr_manager.js

```diff
@@ -184,13 +184,33 @@ shaka.abr.SimpleAbrManager = class {
     if (this.resizeObserver_ && this.config_.restrictToElementSize) {
       const devicePixelRatio =
           this.config_.ignoreDevicePixelRatio ? 1 : window.devicePixelRatio;
-      maxHeight = this.mediaElement_.clientHeight * devicePixelRatio;
-      maxWidth = this.mediaElement_.clientWidth * devicePixelRatio;
+      maxHeight = Math.min(
+          maxHeight, this.mediaElement_.clientHeight * devicePixelRatio);
+      maxWidth = Math.min(
+          maxWidth, this.mediaElement_.clientWidth * devicePixelRatio);
     }
 
     // Get sorted Variants.
     let sortedVariants = SimpleAbrManager.filterAndSortVariants_(
-        this.config_.restrictions, this.variants_, maxHeight, maxWidth);
+        this.config_.restrictions, this.variants_,
+        /* maxHeight= */ Infinity, /* maxWidth= */ Infinity);
+
+    if (maxHeight != Infinity || maxWidth != Infinity) {
+      // let index =
+      const resolutions = SimpleAbrManager.getResolutionList_(sortedVariants);
+      for (const resolution of resolutions) {
+        if (resolution.height >= maxHeight || resolution.width >= maxWidth) {
+          maxHeight = resolution.height;
+          maxWidth = resolution.width;
+          if (resolution.height >= maxHeight && resolution.width >= maxWidth) {
```

Comment:
I don't understand your logic here.  On the previous two lines, we set maxHeight and maxWidth to resolution.height and resolution.width, so this condition is always true on this line.  Right?

Overall, I would find this easier to review and verify with some comments in the code.  Thanks!

Reply:
I added a comment with an example for explanatory purposes.

Reply:
I don't think this code would result in the behavior you explain in the comments, of looking for the smallest resolution that will "fit" the given max resolution. Given that it happens after maxWidth and maxHeight are re-defined, the inner if-condition is really testing `resolution.height >= resolution.height && resolution.width >= resolution.width`, which will only be false if the resolution is `undefined`, and thus it will cause the first resolution where EITHER width or height fits to be picked.

I think you probably just want to make the inner if-clause the outer check, and make it always break. Like...
```
if (resolution.height >= maxHeight && resolution.width >= maxWidth) {
  maxHeight = resolution.height;
  maxWidth = resolution.width;
  break;
}
```

Reply:
Ok.. I changed it!

Reply:
This makes much more sense to me.  Thanks!